haskell-indexer-backend-ghc/src/Language/Haskell/Indexer/Backend/Ghc.hs

```diff
@@ -220,8 +220,22 @@ declsFromRenamed ctx (hsGroup, _, _, _) =
                                 $ hsGroup
         explicitVarBindDefs = map mkDeclName explicitNames
         varBindDefs = explicitVarBindDefs ++ implicitVarBindDefs
-    in (concat [defs, instDefs, varBindDefs], instChanges)
+        valBindDefs = case hs_valds hsGroup of
+            ValBinds _ _ _ ->
+                error "should not hit ValBinds when accessing renamed AST"
+            XValBindsLR (NValBinds binds _sigs) -> collectPatSymDefs binds
+    in (concat [defs, instDefs, varBindDefs, valBindDefs], instChanges)
   where
+    -- TODO(jinwoo): Collect for signatures too.
+    collectPatSymDefs :: [(GHC.RecFlag, LHsBinds GhcRn)] -> [DeclAndAlt]
```

Comment:
I think you'll save some verbiage by doing
concatMap (collectPatBinds . snd) binds